R/descriptives.R

```diff
@@ -673,34 +635,42 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
   if (options$mode) {
 
     # TODO: fix this after we have mixed columns (DvdB)
-    if (FALSE && is.numeric(na.omitted)) { # scale data
+    if (FALSE && columnType == "scale") { # scale data
       temp <- .desriptivesComputeModeContinuous(na.omitted)
       mode <- temp[["xValues"]][which.max(temp[["yValues"]])]
 
       shouldAddModeMoreThanOnceFootnote <- temp[["numModes"]] > 1L
     } else { # ordinal, nominal, or nominal text data
       tb <- table(na.omitted)
-      mode <- names(tb[tb == max(tb)])[1] # use only the first mode
-      mode <- if (!is.na(suppressWarnings(as.numeric(mode)))) {
-        as.numeric(mode) # the most frequent value with mixed columns we can always show this one
-      } else {
-        which.max(tb) # the index of the most frequent value, not great for nominal text but better than nothing?
-      }
+      allModes <- names(tb[tb == max(tb)])
+      mode <- allModes[1] # use only the first mode
 
-      shouldAddModeMoreThanOnceFootnote <- length(mode) > 1L
+      shouldAddModeMoreThanOnceFootnote <- length(allModes) > 1L
     }
 
-    shouldAddModeContinuousTreatedAsDiscreteFootnote <- is.numeric(na.omitted)
+    shouldAddModeContinuousTreatedAsDiscreteFootnote <- columnType == "scale"
 
+    # need some help here @vandenman
+    # modeType <- switch(columnType,
+    #                    "scale" = "number",
+    #                    "ordinal" = "integer",
+    #                    "nominal" = "string")
+    # resultsCol[["Mode"]] <- createMixedRow(
+    #   value = mode[1L],
+    #   type =  modeType
+    # )
     resultsCol[["Mode"]] <- mode[1L]
 
   }
 
   if (options$quartiles) {
-    if (base::is.factor(na.omitted) == FALSE) {
-      resultsCol[["q1"]] <- quantile(na.omitted, c(.25), names = FALSE)
-      resultsCol[["q2"]] <- quantile(na.omitted, c(.5), names = FALSE)
-      resultsCol[["q3"]] <- quantile(na.omitted, c(.75), names = FALSE)
+    if (columnType == "scale" || columnType == "ordinal") {
+      # Type 7: default in R
+      # Type 3: Nearest even order statistic (SAS default till ca. 2010).
+      quartileType <- ifelse(columnType == "scale", 7, 3)
```

Comment:
This is a functional change, not just a rewrite, no? Not a problem, but good to know.

Reply:
This is because using quantile on an ordered factor (rather than using it on as.numeric(orderedFactor)) results in an error in the default quantile call:
```
> quantile(factor(1:5, ordered = TRUE))
Error in quantile.default(factor(1:5, ordered = TRUE)) : 
  'type' must be 1 or 3 for ordered factors
```

Reply:
Shouldn't we then instead use type 1 or 3? I'm not entirely sure what the differences would be compared to doing as.numeric though.